docs/includes/topic_cert-manager.rst

```diff
@@ -1,12 +1,64 @@
+.. _certificate-manager:
 
+Certificate Manager / SSL Offloading
+====================================
 
-- ``cert_manager``: The certificate manager that manages access to certificates and keys for user authentication. Must be set to ``f5_openstack_agent.lbaasv2.drivers.bigip.barbican_cert.BarbicanCertManager``.
+Overview
+--------
 
-- Keystone v2/v3 authentication
+OpenStack's 'Barbican' certificate manager provides a secure location where users can store sensitive information, such as SSH keys, private keys, certificates, and user passwords (referred to as "`secrets`_ " in OpenStack lingo).
+
+The F5® agent uses Barbican certificates to perform :term:`SSL offloading` on BIG-IP®. It allows users to either create a new SSL profile, or to designate an existing `BIG-IP SSL profile`_ as the parent from which client profiles created for LBaaS objects will inherit settings.
+
+In general, SSL offloading frees up server and application capacity for handling traffic by shifting authentication processing from the target server to a designated authentication server. As shown in the diagram, once an admin user has added `secrets`_ to a Barbican container, he can use it to create a :ref:`TLS loadbalancer <Create a tls load balancer>`. After the certificate data is validated, the F5® agent configures the loadbalancer on the BIG-IP.
+
+.. figure:: ../media/LBaaS_cert-mgr_with-legend.jpg
+    :alt: SSL Offloading with OpenStack Barbican, Neutron LBaaSv2, and BIG-IP
+
+    SSL Offloading with OpenStack Barbican, Neutron LBaaSv2, and BIG-IP
+
+Use Case
+--------
+
+When you configure `Client SSL`_ or `Server SSL`_ profiles and assign them to a virtual server, BIG-IP offloads SSL processing from the destination server. This not only conserves resources on the destination servers, but also enables the BIG-IP to customize SSL traffic processing according to your specifications.
+
+`Client SSL`_ is the most common use case; this makes it possible for the BIG-IP to decrypt client requests before sending them on to a server, and encrypt server responses before sending them back to the client.
+
+Prerequisites
+-------------
+
+- Licensed, operational BIG-IP :term:`device`.
+
+- Operational OpenStack cloud (|openstack| release).
+
+- Administrator access to both BIG-IP device(s) and OpenStack cloud.
+
+- F5 LBaaSv2 driver and :ref:`agent <agent:home>` installed on each server for which BIG-IP LTM services are required.
+
+- `OpenStack Barbican <OpenStack Barbican: https://wiki.openstack.org/wiki/Barbican>`_ certificate manager configured and operational.
+
+- Existing `BIG-IP SSL profile <https://support.f5.com/kb/en-us/products/big-ip_ltm/manuals/product/bigip-ssl-administration-12-0-0/5.html#unique_527799714>`_ (*optional*).
+
+
+Caveats
+-------
+
+None.
+
+
+Configuration
+-------------
+
+1. Edit the :ref:`Agent Configuration File`:
+
+.. code-block:: text
+
+    $ sudo emacs /etc/neutron/services/f5/f5-openstack-agent.ini
```

Comment:
In the LbaaSv2 and F5 Agent docs, vi is used as editor, I was wondering if emacs is used here for any specific reason?